edx_notifications/stores/mongo/store_provider.py

```diff
@@ -0,0 +1,141 @@
+import datetime
+import pymongo
+from pymongo import MongoClient
+import pytz
+from edx_notifications import const
+from edx_notifications.exceptions import BulkOperationTooLarge
+from edx_notifications.stores.mongo.mongo_models import MongoUserNotification
+from edx_notifications.stores.sql.store_provider import SQLNotificationStoreProvider
+
+
+class MongoNotificationStoreProvider(SQLNotificationStoreProvider):
+    """
+
+    """
+
+    def __init__(self, **kwargs):
+        super(MongoNotificationStoreProvider, self).__init__(**kwargs)
+        self.client = MongoClient(kwargs.get('host'), kwargs.get('port'))
+        self.db = self.client[kwargs.get('database_name')]
+        self.collection = self.db.user_notification
+        # self.bulk = self.collection.initializeUnorderedBulkOp()
+
+    def _get_prepaged_notifications(self, user_id, filters=None, options=None):
+        """
+        Helper to set up the notifications query before paging
+        is applied. WARNING: This should be used with care and to not
+        iterate over this returned results set. Typically this
+        will just be used to get a count()
+        """
+
+        _filters = filters if filters else {}
+        _options = options if options else {}
+
+        namespace = _filters.get('namespace')
+        read = _filters.get('read', True)
+        unread = _filters.get('unread', True)
+        type_name = _filters.get('type_name')
+        start_date = _filters.get('start_date')
+        end_date = _filters.get('end_date')
+        query_object = {'user_id': user_id}
+        if not read and not unread:
+            raise ValueError('Bad arg combination either read or unread must be set to True')
+
+        if namespace:
+            query_object.update({'user_notification.namespace': namespace})
+
+        if not (read and unread):
+            if read:
+                query_object.update({'user_notification.read_at': {"$ne": None}})
+
+            if unread:
+                query_object.update({'user_notification.read_at': None})
+
+        if type_name:
+            query_object.update({'user_notification.msg_type_name': type_name})
+
+        if start_date:
+            query_object.update({'created': {"gte": start_date}})
+
+        if end_date:
+            query_object.update({'created': {"lte": end_date}})
+
+        collection = self.collection.find(query_object)
```

Comment:
Sorry, just realized: we need to sort by create date DESC here. Also, the interface is supposed to support OFFSET and LIMIT semantics as well. Not sure how well that maps to MongoDB API, but it might have to be done post-processing.

Either way, we'll need to have the first record in this collection be the most recent one. I'll also add a note below about this.